docs/resources/tutorials/differential-expression-analysis.md

```diff
@@ -0,0 +1,458 @@
+
+For a differential expression analysis to be performed, one needs to know the experimental condition for each of the samples. The sample sheet supplied to *nf-core/rnaseq* does not include this information, so we need to generate a sample sheet that can be supplied to the `limma` R package. Our example sample sheet is called `dme_elev_samples.tsv`, and looks like this:
+
+```bash title="dme_elev_samples.tsv"
+sample population temp
+SRR1576457 maine low
+SRR1576458 maine low
+SRR1576459 maine low
+SRR1576460 maine high
+SRR1576461 maine high
+SRR1576462 maine high
+SRR1576463 panama low
+SRR1576464 panama low
+SRR1576465 panama low
+SRR1576514 panama high
+SRR1576515 panama high
+SRR1576516 panama high
+```
+
+### 1. Expression analysis in R: preliminaries
+The differential expression analysis we demonstrate requires a number of R packages, which can be installed as follows:
+
+```R
+install.packages("tidyverse")
+
+if (!require("BiocManager", quietly = TRUE))
+    install.packages("BiocManager")
+
+BiocManager::install("limma","edgeR")
+```
+
+`Tidyverse` is a data science packages that facilitates the manipulation of tabular data, and plotting visualizations. `edgeR` and `limma` are bioinformatics packages for analyzing bulk RNA-seq data.
+
+### 2. Load gene-level abundance data
+The first two columns of `salmon.merged.gene_counts.tsv` are "gene_id", the original gene id in the annotation, and "gene_name", the gene symbol. There may be cases in which a gene symbol is not unique among gene ids, such that it is useful to concatenate these two columns into a new label, which we do here. To manipulate the data into a matrix format that limma expects, we also need to remove the id and name information from the data table, and set the new concatenated label as row names. We do basic table manipulation by loading the table as a tibble via tidyverse, but then, since tidyverse doesn't accomodate row names, convert back to a data frame.
+
+```R
+expression_data <-read_table("salmon.merged.gene_counts.tsv") %>%
+                  mutate(gene_id_symbol=paste(gene_id,gene_name,sep="_")) %>%
+                  select(!c(gene_id,gene_name)) %>%
+                  relocate(gene_id_symbol)
+
+new_ids=expression_data$gene_id_symbol
+expression_data <- expression_data %>% select(!c(gene_id_symbol))
+expression_data <-as.data.frame(expression_data)
+row.names(expression_data)=new_ids
+```
```

Comment:
if we drop the merging gene id and gene symbol don't need to do the tibble -> data frame -> tibble, can just read in the data with row names